dask_planner/src/sql/types.rs

```diff
@@ -269,58 +270,85 @@ impl SqlTypeName {
-            "SYMBOL" => SqlTypeName::SYMBOL,
-            "DECIMAL" => SqlTypeName::DECIMAL,
-            "DYNAMIC_STAT" => SqlTypeName::DYNAMIC_STAR,
-            "UNKNOWN" => SqlTypeName::UNKNOWN,
-            _ => unimplemented!("SqlTypeName::from_string() for str type: {}", input_type),
+    pub fn from_string(input_type: &str) -> PyResult<Self> {
+        match input_type.to_uppercase().as_ref() {
+            "ANY" => Ok(SqlTypeName::ANY),
+            "ARRAY" => Ok(SqlTypeName::ARRAY),
+            "NULL" => Ok(SqlTypeName::NULL),
+            "BOOLEAN" => Ok(SqlTypeName::BOOLEAN),
+            "COLUMN_LIST" => Ok(SqlTypeName::COLUMN_LIST),
+            "DISTINCT" => Ok(SqlTypeName::DISTINCT),
+            "CURSOR" => Ok(SqlTypeName::CURSOR),
+            "TINYINT" => Ok(SqlTypeName::TINYINT),
+            "SMALLINT" => Ok(SqlTypeName::SMALLINT),
+            "INT" => Ok(SqlTypeName::INTEGER),
+            "INTEGER" => Ok(SqlTypeName::INTEGER),
+            "BIGINT" => Ok(SqlTypeName::BIGINT),
+            "REAL" => Ok(SqlTypeName::REAL),
+            "FLOAT" => Ok(SqlTypeName::FLOAT),
+            "GEOMETRY" => Ok(SqlTypeName::GEOMETRY),
+            "DOUBLE" => Ok(SqlTypeName::DOUBLE),
+            "TIME" => Ok(SqlTypeName::TIME),
+            "TIME_WITH_LOCAL_TIME_ZONE" => Ok(SqlTypeName::TIME_WITH_LOCAL_TIME_ZONE),
+            "TIMESTAMP" => Ok(SqlTypeName::TIMESTAMP),
+            "TIMESTAMP_WITH_LOCAL_TIME_ZONE" => Ok(SqlTypeName::TIMESTAMP_WITH_LOCAL_TIME_ZONE),
+            "DATE" => Ok(SqlTypeName::DATE),
+            "INTERVAL" => Ok(SqlTypeName::INTERVAL),
+            "INTERVAL_DAY" => Ok(SqlTypeName::INTERVAL_DAY),
+            "INTERVAL_DAY_HOUR" => Ok(SqlTypeName::INTERVAL_DAY_HOUR),
+            "INTERVAL_DAY_MINUTE" => Ok(SqlTypeName::INTERVAL_DAY_MINUTE),
+            "INTERVAL_DAY_SECOND" => Ok(SqlTypeName::INTERVAL_DAY_SECOND),
+            "INTERVAL_HOUR" => Ok(SqlTypeName::INTERVAL_HOUR),
+            "INTERVAL_HOUR_MINUTE" => Ok(SqlTypeName::INTERVAL_HOUR_MINUTE),
+            "INTERVAL_HOUR_SECOND" => Ok(SqlTypeName::INTERVAL_HOUR_SECOND),
+            "INTERVAL_MINUTE" => Ok(SqlTypeName::INTERVAL_MINUTE),
+            "INTERVAL_MINUTE_SECOND" => Ok(SqlTypeName::INTERVAL_MINUTE_SECOND),
+            "INTERVAL_MONTH" => Ok(SqlTypeName::INTERVAL_MONTH),
+            "INTERVAL_SECOND" => Ok(SqlTypeName::INTERVAL_SECOND),
+            "INTERVAL_YEAR" => Ok(SqlTypeName::INTERVAL_YEAR),
+            "INTERVAL_YEAR_MONTH" => Ok(SqlTypeName::INTERVAL_YEAR_MONTH),
+            "MAP" => Ok(SqlTypeName::MAP),
+            "MULTISET" => Ok(SqlTypeName::MULTISET),
+            "OTHER" => Ok(SqlTypeName::OTHER),
+            "ROW" => Ok(SqlTypeName::ROW),
+            "SARG" => Ok(SqlTypeName::SARG),
+            "BINARY" => Ok(SqlTypeName::BINARY),
+            "VARBINARY" => Ok(SqlTypeName::VARBINARY),
+            "CHAR" => Ok(SqlTypeName::CHAR),
+            "VARCHAR" | "STRING" => Ok(SqlTypeName::VARCHAR),
```

Comment:
I've just tried these changes with my own tests.

I can now use "string" types, but varchars w/ a defined limit still fail:
```
Traceback (most recent call last):               
  File "<stdin>", line 1, in <module>                                                                            
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/context.py", line 238, in create_table
    dc = InputUtil.to_dc(                                                                                                   
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 68, in to_dc
    table = filled_get_dask_dataframe(input_item)                                                           
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 57, in <lambda>
    filled_get_dask_dataframe = lambda *args: cls._get_dask_dataframe(                                                      
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 90, in _get_dask_dataframe
    return plugin.to_dc(                                                                               
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/hive.py", line 69, in to_dc
    column_information = {                                                                                    
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/hive.py", line 70, in <dictcomp>
    col: sql_to_python_type(SqlTypeName.fromString(col_type.upper()))                                      
RuntimeError: Internal("Cannot determine SQL type name for 'VARCHAR(65535)'")
```

Reply:
I have now pushed changes to support `VARCHAR(n)` and some other parameterized types